src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -1660,6 +1660,60 @@ public string RegularProperty2
                 );
         }
 
+        [Fact, WorkItem(50053, "https://github.com/dotnet/roslyn/issues/50053")]
+        public void PrivatelyImplementingInitOnlyProperty_ReferenceConversion()
+        {
+            string source = @"
+var x = new DerivedType() { SomethingElse = 42 };
+System.Console.Write(x.SomethingElse);
+
+public interface ISomething { int Property { get; init; } }
+public record BaseType : ISomething { int ISomething.Property { get; init; } }
+
+public record DerivedType : BaseType
+{
+    public int SomethingElse
+    {
+        get => ((ISomething)this).Property;
+        init => ((ISomething)this).Property = value;
```

Comment:
>this [](start = 29, length = 4)

Consider adding a test with `base`.

Reply:
Also a test where `this` hides an `init`-only property from `base`.

---
In reply to: [558461131](https://github.com/dotnet/roslyn/pull/50424#discussion_r558461131) [](ancestors = 558461131)